File: Python/7py 2/mod256.py
"""
|--------------------------------------------------------------------------
| MOD 256 without the MOD operator
|--------------------------------------------------------------------------
|
"""

def mod256_without_mod(number):
   return  number - ((number // 256) * 256)


def mod256_without_mod(number):
   return number & 255


def mod256_without_mod(number):
   return pow(number, 1, 256)


print(mod256_without_mod(254), 254)
print(mod256_without_mod(256), 0)
print(mod256_without_mod(258), 2)
print(mod256_without_mod(-254), 2)
print(mod256_without_mod(-256), 0)
print(mod256_without_mod(-258), 254)

File: JavaScript/7js 2/mod256.js
/*
|--------------------------------------------------------------------------
| MOD 256 without the MOD operator
|--------------------------------------------------------------------------
|
*/

function mod256WithoutMod(number) {
   return number - ((number / 256) ^ 0) * 256;
}


function mod256WithoutMod(number) {
   return number - ((number / 256) | 0) * 256;
}


function mod256WithoutMod(number) {
   return number - ~~(number / 256) * 256;
}


console.log(mod256WithoutMod(254), 254);
console.log(mod256WithoutMod(256), 0);
console.log(mod256WithoutMod(258), 2);
console.log(mod256WithoutMod(-254), -254);
console.log(mod256WithoutMod(-256), 0);
console.log(mod256WithoutMod(-258), -2);